modules/JsonExporter/src/main/java/uk/ac/ox/oii/jsonexporter/JSONExporter.java

```diff
@@ -0,0 +1,277 @@
+
+
+                    GraphEdge jEdge = new GraphEdge(String.valueOf(e.getId()));
+                    jEdge.setSource(sourceId);
+                    jEdge.setTarget(targetId);
+                    jEdge.setSize(e.getWeight());
+
+                    boolean mixColors=false;
+                    String color="";
+                    jEdge.setLabel(e.getLabel());
+
+                    float r=e.r();
+                    float g=e.g();
+                    float b=e.b();
+
+                    if (r==-1 || g==-1 || b==-1) {
+                        //Mix colors
+                        mixColors=true;
+                    } else {
+                        color = "rgb(" + (int) (r* 255) + "," + (int) (g* 255) + "," + (int) (b* 255) + ")";
+                    }
+
+                    Iterator<Column> eAttr = e.getAttributeColumns().iterator();
+                    while (eAttr.hasNext()) {
+                        Column col = eAttr.next();
+                        if (col == null) {
+                            continue;
+                        }
+                        String name = col.getTitle();
+                        if (name.equalsIgnoreCase("Id") || name.equalsIgnoreCase("Label")
+                                || name.equalsIgnoreCase("uid")) {
+                            continue;
+                        }
+
+                        Object valObj = e.getAttribute(col);
+                        if (valObj == null) {
+                            continue;
+                        }
+                        String val = valObj.toString();
+                        jEdge.putAttribute(name, val);
+                    }
+                    
+                    if (mixColors) {
+                        //Source
+                        Node n = e.getSource();
+                        Color source = new Color(n.r(),n.g(),n.b());
+                        n = e.getTarget();
+                        Color target = new Color(n.r(),n.g(),n.b());
+                        Color result = colorMixer.getColor(null, source, target);
+                        color = "rgb(" + result.getRed() + "," + result.getGreen() + "," + result.getBlue() + ")";
+                    }
+                    jEdge.setColor(color);
+
+                    jEdges.add(jEdge);
+
+                    if (cancel) {
+                        return false;
+                    }
+                    Progress.progress(progress);
+                }
+
+
+                HashMap<String, HashSet<GraphElement>> json = new HashMap<String, HashSet<GraphElement>>();
+                json.put("nodes", jNodes);
+                json.put("edges", jEdges);
+
+                gson.toJson(json, writer);
+
+                //Finish progress
+                Progress.finish(progress);
+                return true;
+            } else {
+                throw new FileNotFoundException("Writer is null");
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
This is a cleaner way to propagate Exception (i.e. the user will see it)

Logger.getLogger(JSONExporter.class.getName()).log(Level.SEVERE, null, e);

You can remove the throw new RuntimeException() then